src/test/java/com/jcabi/log/SupplierLoggerTest.java

```diff
@@ -0,0 +1,278 @@
+            Matchers.isEmptyString()
+        );
+    }
+
+    /**
+     * SupplierLogger can log a message with warn level.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void warnIsEnabled() throws Exception {
+        final String loggerName = "enwarn";
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger
+            .getLogger(loggerName);
+        final UnitTestAppender app = new UnitTestAppender();
+        app.setName("warnapp");
+        app.activateOptions();
+        logger.addAppender(app);
+        final String text = "test6";
+        logger.setLevel(Level.WARN);
+        Logger.withSupplier().warn(
+            loggerName, "Warn enabled: %s",
+            new Supplier<String>() {
+                public String get() {
+                    return text;
+                }
+            }
+        );
+        MatcherAssert.assertThat(
+            new String(app.output().toByteArray()),
+            Matchers.containsString(text)
+        );
+    }
+
+    /**
+     * SupplierLogger can tell if info is disabled and the message is not
+     * logged enabled.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void infoIsDisabled() throws Exception {
+        final String loggerName = "noinfo";
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger
+            .getLogger(loggerName);
+        final UnitTestAppender app = new UnitTestAppender();
+        app.setName("noinfoapp");
+        app.activateOptions();
+        logger.addAppender(app);
+        logger.setLevel(Level.OFF);
+        Logger.withSupplier().info(
+            loggerName, "Info disabled: %s",
+            new Supplier<String>() {
+                public String get() {
+                    return "test7";
+                }
+            }
+        );
+        MatcherAssert.assertThat(
+            new String(app.output().toByteArray()),
+            Matchers.isEmptyString()
+        );
+    }
+
+    /**
+     * SupplierLogger can log a message with info level.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void infoIsEnabled() throws Exception {
+        final String loggerName = "withinfo";
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger
+            .getLogger(loggerName);
+        final UnitTestAppender app = new UnitTestAppender();
+        app.setName("infoapp");
+        app.activateOptions();
+        logger.addAppender(app);
+        final String text = "text8";
+        logger.setLevel(Level.INFO);
+        Logger.withSupplier().info(
+            loggerName, "Info enabled: %s",
+            new Supplier<String>() {
+                public String get() {
+                    return text;
+                }
+            }
+        );
+        MatcherAssert.assertThat(
+            new String(app.output().toByteArray()),
+            Matchers.containsString(text)
```

Comment:
@mkordas Can we not leave it like this? It's not only pattern change, it's also new variables and whatnot, since qulice will complain about string concatenation, duplicated strings and I don't know what else. 

First off, you don't have only "text8" logged here (so you could check only ``equalsTo(text)``: you have "Info enabled: text8".

Having a simple ``equalsTo(text)`` would mean using only ``%s`` as second param when building the message - then also it's duplicated string and needs variables.

I consider contains more than sufficient since there are **unique strings** across all tests. So there is no chance that one tests logs the same content + every test runs with a new and clean appender, so there is also no chance of having a logged message from an older test run.